drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+{
+	struct bmm350_data *drv_data = dev->data;
+	struct bmm350_mag_temp_data mag_temp_data;
+
+	if (bmm350_get_compensated_mag_xyz_temp_data_fixed(dev, &mag_temp_data) < 0) {
+		LOG_ERR("failed to read sample");
+		return -EIO;
+	}
+	drv_data->mag_temp_data.x = mag_temp_data.x;
+	drv_data->mag_temp_data.y = mag_temp_data.y;
+	drv_data->mag_temp_data.z = mag_temp_data.z;
+	return 0;
+}
+
+/*
+ * ut change to Gauss
+ */
+static void bmm350_convert(struct sensor_value *val, int raw_val)
+{
+	val->val1 = raw_val / 100;
+	val->val2 = (raw_val % 100) * 10000;
+}
+
+static int bmm350_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct bmm350_data *drv_data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_MAGN_X:
+		bmm350_convert(val, drv_data->mag_temp_data.x);
+		break;
+	case SENSOR_CHAN_MAGN_Y:
+		bmm350_convert(val, drv_data->mag_temp_data.y);
+		break;
+	case SENSOR_CHAN_MAGN_Z:
+		bmm350_convert(val, drv_data->mag_temp_data.z);
+		break;
+	case SENSOR_CHAN_MAGN_XYZ:
+		bmm350_convert(val, drv_data->mag_temp_data.x);
+		bmm350_convert(val + 1, drv_data->mag_temp_data.y);
+		bmm350_convert(val + 2, drv_data->mag_temp_data.z);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+static uint8_t acc_odr_to_reg(const struct sensor_value *val)
+{
+	double odr = sensor_value_to_double((struct sensor_value *)val);
+
+	uint8_t reg = BMM350_DATA_RATE_100HZ;
+
+	if ((odr >= 0.78125) && (odr <= 1.5625)) {
+		reg = BMM350_DATA_RATE_1_5625HZ;
+	} else if ((odr > 1.5625) && (odr <= 3.125)) {
+		reg = BMM350_DATA_RATE_3_125HZ;
+	} else if ((odr > 3.125) && (odr <= 6.25)) {
+		reg = BMM350_DATA_RATE_6_25HZ;
+	} else if ((odr > 6.25) && (odr <= 12.5)) {
+		reg = BMM350_DATA_RATE_12_5HZ;
+	} else if ((odr > 12.5) && (odr <= 25.0)) {
+		reg = BMM350_DATA_RATE_25HZ;
+	} else if ((odr > 25.0) && (odr <= 50.0)) {
+		reg = BMM350_DATA_RATE_50HZ;
+	} else if ((odr > 50.0) && (odr <= 100.0)) {
+		reg = BMM350_DATA_RATE_100HZ;
+	} else if ((odr > 100.0) && (odr <= 200.0)) {
+		reg = BMM350_DATA_RATE_200HZ;
+	} else if ((odr > 200.0) && (odr <= 400.0)) {
+		reg = BMM350_DATA_RATE_400HZ;
+	} else if (odr > 400.0) {
+		reg = BMM350_DATA_RATE_400HZ;
+	}
+	return reg;
+}
+
+static uint8_t acc_osr_to_reg(const struct sensor_value *val)
+{
+	switch (val->val1) {
+	case 0:
+		return BMM350_NO_AVERAGING;
+	case 2:
+		return BMM350_AVERAGING_2;
+	case 4:
+		return BMM350_AVERAGING_4;
+	case 8:
+		return BMM350_AVERAGING_8;
+	default:
+		return 0xFF;
+	}
+}
+
+/*!
+ * @brief This API sets the ODR and averaging factor.
+ */
+int8_t bmm350_set_odr_performance(enum bmm350_data_rates odr,
+				  enum bmm350_performance_parameters performance,
+				  const struct device *dev)
+{
+	/* Variable to store the function result */
+	int8_t rslt = 0x00;
+	/* Variable to get PMU command */
+	uint8_t reg_data = 0;
+	enum bmm350_performance_parameters performance_fix = performance;
+
+	if (rslt == BMM350_OK) {
```

Comment:
this if is weird, rslt = 0x00 only 5 lines above and never set otherwise, why the branch here?

Reply:
you are correct, I will remove the redundant if.